docs/source/reference/kubernetes/index.rst

```diff
@@ -0,0 +1,154 @@
+.. _kubernetes-overview:
+
+SkyPilot on Kubernetes (Alpha)
+==============================
+
+.. note::
+    Kubernetes support for SkyPilot is an alpha preview under active development.
+    There may be rough edges and features may change without notice.
+    Please report any `bugs <https://github.com/skypilot-org/skypilot/issues>`_ and
+    `reach out to us <http://slack.skypilot.co>`_ for feature requests.
+
+SkyPilot tasks can be deployed on your private on-prem or cloud clusters running Kubernetes.
+The Kubernetes cluster gets added to the list of "clouds" in SkyPilot and SkyPilot
+tasks can be submitted to your Kubernetes cluster just like any other cloud provider.
+
+**Benefits of bringing your Kubernetes cluster to SkyPilot:**
+
+* Get SkyPilot features (setup management, job execution, queuing, logging, SSH access) on your Kubernetes resources
+* Replace complex Kubernetes manifests with simple SkyPilot tasks
+* Maximize resource utilization by running cloud jobs on your Kubernetes cluster.
+* Seamlessly "burst" jobs to the cloud if the Kubernetes cluster is congested.
+* Retain observability and control over your cluster with your existing Kubernetes tools.
+
+**Supported deployment models:**
+
+* On-prem clusters (Kubeadm, K3s, Rancher)
+* Hosted Kubernetes services (AWS EKS, GKE)
+* Local development clusters (KinD, minikube)
+
+
+Kubernetes Cluster Requirements
+-------------------------------
+
+To connect and use a Kubernetes cluster, SkyPilot needs:
+
+* An existing Kubernetes cluster running Kubernetes v1.20 or later.
+* A `Kubeconfig <kubeconfig>`_ file containing access credentials and namespace to be used.
+
+In a typical workflow:
+
+1. A cluster administrator sets up a Kubernetes cluster. Detailed admin guides for
+   different deployment environments (Amazon EKS, Google GKE, On-Prem and local debugging) are included in the :ref:`Kubernetes cluster setup guide <kubernetes-setup>`.
+
+2. Users who want to run SkyPilot tasks on this cluster are issued Kubeconfig
+   files containing their credentials (`kube-context <https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/#define-clusters-users-and-contexts>`_).
+   SkyPilot reads this Kubeconfig file to communicate with the cluster.
+
+Submitting SkyPilot tasks to Kubernetes Clusters
+------------------------------------------------
+.. _kubernetes-instructions:
+
+Once your cluster administrator has :ref:`setup a Kubernetes cluster <kubernetes-setup>` and provided you with a kubeconfig file:
+
+0. Make sure `kubectl <https://kubernetes.io/docs/tasks/tools/>`_, ``socat`` and ``lsof`` are installed.
```

Comment:
Where? on local machine? My mac doesn't have `socat` but it ran k8s before.

Reply:
Ah thanks - clarified for local machine now. `socat` and `lsof` are dependencies introduced with #2500.

Reply:
Not sure what the best way but wearing a user's hat, having `socat` as a requirement is barrier to entry? Mac doesn't come with it. Can we skip it from docs?

Reply:
It's required, so cannot be removed from docs. I think it's a fairly low barrier to entry, and about as hard as installing kubectl (i.e., one line `brew install kubectl socat`). I've updated the docs to include commands for installing these dependencies. 

Our code also includes pre-flight checks for these deps, and if not present throws a nice error suggesting how to install.